packages/grid/x-grid/src/XGrid.test.tsx

```diff
@@ -264,4 +265,59 @@ describe('<XGrid />', () => {
       expect(getColumnValues()).to.deep.equal(['Puma', 'Nike', 'Adidas']);
     });
   });
+  describe('state', () => {
+    it('should allow to control the state using useState', () => {
+      function GridStateTest() {
+        const [gridState] = useState<Partial<GridState>>({
+          sorting: { sortModel: [{ field: 'brand', sort: 'desc' }], sortedRows: [2, 0, 1] },
+        });
+
+        return (
+          <div style={{ width: 300, height: 300 }}>
+            <XGrid {...defaultProps} state={gridState} />
+          </div>
+        );
+      }
+
+      render(<GridStateTest />);
+      expect(getColumnValues()).to.deep.equal(['Puma', 'Nike', 'Adidas']);
+    });
```

Comment:
I think that we should add a second `setProps({  })` call to guarantee that the logic can handle rerenders, not only the first render.